### ISSUE TITLE
Mentors should be able to mentor for multiple Industries and indicate their stage of career for each one

### ISSUE THREAD

**Issue by rebeccafae** [womenrising/womenrising#88]
When a user signs up to be a mentor, allow them to choose multiple industries, and select a stage of career for each.

(checkbox for each industry, with a dropdown for the career level if they check the box)

**Comment by KateCatlin** [comment]
@danrice92 is going to work on this one!

**Comment by danrice92** [comment]
Hi folks, I'm going to start working on this one as my first contribution in my fork of the project.

**Comment by danrice92** [comment]
Hey Rachel, Kate,

I've been working through this issue for a while and it seems like there are a couple different approaches to solving it. Right now there are mentor_industry and stage_of_career constants in the User model, with just a string and integer in the users table of the database to reference them.

It would be faster and easier now to just shove comma-delimited strings and integers into the database and I could probably have this issue resolved pretty quickly, but eventually this could cause some problems down the road if you want to add more industry or career stage options if the client-side code sends the wrong information into the database.

Alternately, if you prefer, I could make the changes to the database that would make a mentor_industries table with career stages so it'll be easier to maintain in the future, but it'll probably be a longer/harder issue to tackle and will probably require more testing before the production database uses this pull to avoid data loss. Is there a way you guys would prefer it? If you'd rather this be done faster and not have too many database changes made, I could do it that way, or I could go for the harder (but probably better, in the long run) approach.

Keep me posted!

**Comment by rebeccafae** [comment]
Hey Dan,
(my name is Rebecca 😅 )

I was talking to @marshallds about this. He has a good idea for an approach and should be commenting on this soon.

Thanks for working on this!!!

**Comment by marshallds** [comment]
Hi @danrice92 
I was talking to @rfjackson11 about this one and as you point out the "correct" database way of handling this is a bit involved. We would need a table for mentor_industries and a join table to members that had a career level field. Rails makes this less bad but it's still a bit involved.  Also, if a mentor_industry was ever deleted there would be a cascade problem (everyone who had that selected would lose it).

I don't like comma delimited strings in a database because I've seen parse errors far too often and they don't query well--and we will want to query on this stuff! But there is a better version of this that might be good. [Postgres supports a json field](https://blog.codeship.com/unleash-the-power-of-storing-json-in-postgres/) that could hold the information in a nicely structured way. And you can query and index it!  The downsides are the json query syntax is a bit obscure so we'd want to write a couple of methods or scopes to encapsulate it and we also don't get a lot of help from the built in rails form helpers.  It's a bit of a weird form situation anyway though.

I'd be happy to talk to you more about the details if you like.  Might be helpful to whiteboard or something though.

**Comment by danrice92** [comment]
@rfjackson11 @marshallds

Ah! Rebecca, sorry about that! I will one day find a way to remember names better, my bad. It'll help when I've said more than like three words to everyone. :)

Hey Marshall, I was talking to my Bloc mentor a bit about how to approach this sort of issue given how the database is now. He made pretty much the same recommendation involving a join table and mentor_industries being its own table rather than just a variable- I just noticed that leaving it as it is and manipulating the string was likely to get messy, especially since Women Rising already has production data, and since I frankly didn't know about the json support in Postgres at all, I figured a mentor_industries table would be a good solution.

Any chance you'd have some time tomorrow morning before noon, Thursday before 9, or Thursday after 6 to talk about it at Radial? (working around T-Mobile schedule). We can message through here too but it's a bit slower, I suppose. Or I'm on the Equality Hackers Slack!

Sorry for the short notice if tomorrow is the best of those three times- just got off work today, as a matter of fact. But I'd love to chat more about it!
